src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,120 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+/**
+ * A storage handler that loads/saves tasks to the file.
+ */
+public class Storage {
+    private final String FILEPATH;
+
+    public Storage(String FILEPATH) {
+        this.FILEPATH = FILEPATH;
+    }
+
+    /**
+     * Reads tasks from storage and returns them in an ArrayList.
+     *
+     * @return An ArrayList of Tasks.
+     * @throws FileNotFoundException If the storage file cannot be found.
+     */
+    public ArrayList<Task> load() throws FileNotFoundException {
+        File f = new File(FILEPATH);
+        f.getParentFile().mkdirs();
+        try {
+            if (!f.exists()) {
+                f.createNewFile();
+            }
+        } catch(IOException e) {
+            System.out.println("Unable to read file.");
+        }
+
+        ArrayList<Task> tasks = new ArrayList<>();
+        Scanner s = new Scanner(f);
+        while(s.hasNext()) {
+            String nextTask = s.nextLine();
+            String[] temp = nextTask.split("\\[");
+            Character taskType = temp[1].charAt(0);
+            Boolean marked = temp[2].charAt(0) == 'X';
+            String dateTemp = temp[temp.length - 1];
+            dateTemp = dateTemp.substring(0, dateTemp.length() - 1);
+            LocalDate date = LocalDate.now();
```

Comment:
This portion of the code can be a bit hard to read and understand, it would be nice to include some logical spaces and comments to help the read understand it more easily